File: LivrusVirtus/Delphi5/ADO/fontes/CadAutores.pas
unit CadAutores;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Cadastros, Db, DBTables, StdCtrls;

type
  TfrmCadAutores = class(TfrmCadastros)
    procedure btnNovoClick(Sender: TObject);
    procedure btnExcluirClick(Sender: TObject);
    procedure btnPesquisarClick(Sender: TObject);
    procedure btnSalvarClick(Sender: TObject);
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
  private
    { Private declarations }
  protected
    { protected declarations }
    function ValidaDados(bTodosDados: boolean): boolean; override;
  public
    { Public declarations }
  end;

var
  frmCadAutores: TfrmCadAutores;

implementation

uses RotinasGlobais, ConsultasSQL, Principal;

{$R *.DFM}

function TfrmCadAutores.ValidaDados(bTodosDados: boolean): boolean;
begin
result:=true;
if Rotinas.ValidaCampo(edtCodigo.Text,'Código não informado!') then begin
  result:=false;
  exit;
end;
if bTodosDados then begin
  if Rotinas.ValidaCampo(edtDescricao.Text,'Autor não informado!') then begin
    result:=false;
    exit;
  end;
end;
end;

procedure TfrmCadAutores.btnNovoClick(Sender: TObject);
begin
  inherited;
NovoDado(ConsSQL.Autor('N',edtCodigo.Text,''));
end;

procedure TfrmCadAutores.btnExcluirClick(Sender: TObject);
begin
  inherited;
ExcluirDados(ConsSQL.Autor('D',edtCodigo.Text,''));
end;

procedure TfrmCadAutores.btnPesquisarClick(Sender: TObject);
begin
  inherited;
PesquisarDados(ConsSQL.Autor('S',edtCodigo.Text,''),
'Autor não encontrado!');
end;

procedure TfrmCadAutores.btnSalvarClick(Sender: TObject);
begin
  inherited;
SalvarDados(ConsSQL.Autor('S',edtCodigo.Text,''),
ConsSQL.Autor('U',edtCodigo.Text,edtDescricao.Text),
ConsSQL.Autor('I',edtCodigo.Text,edtDescricao.Text),true);
end;

procedure TfrmCadAutores.FormClose(Sender: TObject;
  var Action: TCloseAction);
begin
  frmPrincipal.Autores1.Enabled:=true;
  inherited;
end;

end.

File: LivrusVirtus/CPPBuilder6/CLX/dbExpress/fontes/CadAutores.cpp
//---------------------------------------------------------------------------

#include <clx.h>
#pragma hdrstop

#include "CadAutores.h"
#include "ConsultasSQL.h"
#include "Principal.h"
#include "RotinasGlobais.h"
//---------------------------------------------------------------------------
#pragma link "Cadastros"
#pragma resource "*.xfm"
TfrmCadAutores *frmCadAutores;
//---------------------------------------------------------------------------
__fastcall TfrmCadAutores::TfrmCadAutores(TComponent* Owner)
        : TfrmCadastros(Owner)
{
}
//---------------------------------------------------------------------------
Boolean TfrmCadAutores::ValidaDados(Boolean bTodosDados)
{
if (Rotinas->ValidaCampo(edtCodigo->Text,"Código não informado!"))
  return false;
if (bTodosDados) {
  if (Rotinas->ValidaCampo(edtDescricao->Text,"Autor não informado!"))
    return false;
}
return true;
}
//---------------------------------------------------------------------------
void __fastcall TfrmCadAutores::btnNovoClick(TObject *Sender)
{
TfrmCadastros::btnNovoClick(Sender);
NovoDado(ConsSQL->Autor('N',edtCodigo->Text,""));
}
//---------------------------------------------------------------------------

void __fastcall TfrmCadAutores::btnExcluirClick(TObject *Sender)
{
ExcluirDados(ConsSQL->Autor('D',edtCodigo->Text,""));
}
//---------------------------------------------------------------------------

void __fastcall TfrmCadAutores::btnPesquisarClick(TObject *Sender)
{
PesquisarDados(ConsSQL->Autor('S',edtCodigo->Text,""),
"Autor não encontrado!");
}
//---------------------------------------------------------------------------

void __fastcall TfrmCadAutores::btnSalvarClick(TObject *Sender)
{
SalvarDados(ConsSQL->Autor('S',edtCodigo->Text,""),
ConsSQL->Autor('U',edtCodigo->Text,edtDescricao->Text),
ConsSQL->Autor('I',edtCodigo->Text,edtDescricao->Text),true);
}
//---------------------------------------------------------------------------

void __fastcall TfrmCadAutores::FormClose(TObject *Sender,
      TCloseAction &Action)
{
  frmPrincipal->Autores1->Enabled=true;
  TfrmCadastros::FormClose(Sender,Action);
}
//---------------------------------------------------------------------------

